eureka/lib/util.py

```diff
@@ -32,32 +32,32 @@ def trim(data, meta):
 
     Parameters
     ----------
-    data:   DataClass
-        The data object.
+    data:   Xarray Dataset
+        The Dataset object.
     meta:   MetaClass
         The metadata object.
 
     Returns
     -------
-    data:   DataClass
-        The data object with added subdata arrays with trimmed edges depending on xwindow and ywindow which have been set in the S3 ecf.
+    subdata:   Xarray Dataset
+        A new Dataset object with arrays that have been trimmed, depending on xwindow and ywindow as set in the S3 ecf.
     meta:   MetaClass
         The metadata object.
     """
-    data.subdata = data.data[:, meta.ywindow[0]:meta.ywindow[1], meta.xwindow[0]:meta.xwindow[1]]
-    data.suberr  = data.err[:, meta.ywindow[0]:meta.ywindow[1], meta.xwindow[0]:meta.xwindow[1]]
-    data.subdq   = data.dq[:, meta.ywindow[0]:meta.ywindow[1], meta.xwindow[0]:meta.xwindow[1]]
-    data.subwave = data.wave[meta.ywindow[0]:meta.ywindow[1], meta.xwindow[0]:meta.xwindow[1]]
-    data.subv0   = data.v0[:, meta.ywindow[0]:meta.ywindow[1], meta.xwindow[0]:meta.xwindow[1]]
+    subdata = data.isel(y=np.arange(meta.ywindow[0],meta.ywindow[1]),
+                        x=np.arange(meta.xwindow[0],meta.xwindow[1]))
+    # Trim wavelength attribute
+    # data.attrs['wave_2d'] = data.attrs['wave_2d'][meta.ywindow[0]:meta.ywindow[1],
+    #                                             meta.xwindow[0]:meta.xwindow[1]]
```

Comment:
Does the xarray isel magic handle the trimming of wavelengths and other arrays? If so, we can delete the commented out code

Reply:
Yes, it trims all arrays within the Dataset